ui/imports/utils/SubscriptionBroker.qml

```diff
@@ -0,0 +1,196 @@
+import QtQuick 2.15
+
+import utils 1.0
+
+//This is a helper component that is used to batch requests and send them periodically
+//It is used to reduce the number of requests sent to the server and notify different subscribers of the same request
+//It is used by the Subscription component
+QtObject {
+    id: root
+
+    signal request(string topic)
+    function response(topic, responseObj) {
+        d.onResponse(topic, responseObj)
+    }
+    function subscribe(subscription) {
+        d.subscribe(subscription)
+    }
+    function unsubscribe(subscription) {
+        d.unsubscribe(subscription)
+    }
+
+
+    readonly property QtObject d: QtObject {
+        //Mapping subscriptionId to subscription object
+        //subscriptionId is a string and represents the id of the subscription
+        //E.g. "subscriptionId": {subscription: subscriptionObject, topic: "topic"}
+        //The purpose of this mapping is to keep track of the subscriptions and their topics
+        readonly property var managedSubscriptions: ({})
+
+        //Mapping topic to subscriptionIds and request data
+        //topic is a string and represents the topic of the subscription
+        //E.g. "topic": {requestTimestamp: 0, requestInterval: 1000, subscriptions: ["subscriptionId1", "subscriptionId2"], response: null}
+        readonly property var topics: ({})
+        property int topicsCount: 0 //helper property to track change events
+
+        readonly property int requestInterval: {
+            //dependency:
+            d.topicsCount
+            const topicInfos = Object.values(d.topics)
+            if(!topicsCount || topicInfos.length === 0) return 0
+            
+            return topicInfos.reduce(function(minInterval, topicInfo) { return Math.min(minInterval, topicInfo.requestInterval) }, Number.MAX_SAFE_INTEGER)
+        }
+
+        readonly property Timer requestTimer: Timer {
+            interval: d.requestInterval
+            repeat: true
+            running: interval > 0 && Global.applicationWindow.active
```

Comment:
As `SubscriptionBroker` is intended to be generic tool, maybe it would be better to expose `property bool active` and allow users to decide about activity criteria instead of sticking to a specific criterion.

Reply:
Done. I've added the `active` property and it's set to Global.applicationWindow.active in `TransactionFeesBroker.qml` where SubscriptionBroker is used. `Window.active` is not available because the `TransactionFeesBroker.qml` is not an `Item`